jina/executors/indexers/cache.py

```diff
@@ -109,14 +125,13 @@ def delete(self, keys: Iterable[str], *args, **kwargs) -> None:
             self._size -= 1
 
     def get_add_handler(self):
-        # not needed, as we use the queryhandler
-        # FIXME better way to silence warnings
-        return 1
+        """Get the CacheHandler."""
+        return self.get_query_handler()
 
     def get_query_handler(self) -> CacheHandler:
-        return self.CacheHandler(self.index_abspath, self.logger)
+        """Get the CacheHandler."""
+        return self.CacheHandler(self.save_abspath, self.logger)
 
     def get_create_handler(self):
-        # not needed, as we use the queryhandler
-        # FIXME better way to silence warnings
-        return 1
+        """Get the CacheHandler."""
```

Comment:
can we then use the proper access handler depending on the operation?

Reply:
No point. It would just make it more confusing, I think. We only need one.